### PR TITLE
publish workflow to use reusable action

### PR DESCRIPTION
**Pull request by AnieeG**
## Solution

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -92,25 +65,6 @@ jobs:
...
-          aws-region: ${{ env.AWS_REGION }}
-      - name: Login to ECR
-        uses: docker/login-action@42d299face0c5c43a0487c477f595ac9cf22f1a7 # v1.12.0
-        with:
-          registry: ${{ env.AWS_ECR_REPO_PUBLIC_REGISTRY }}
-      # We use the core SemVer version and then append the the CCIP version in
-      # a git tag like: v2.5.0-ccip1.1.1
-      # We also remove the leading "v" prefix from the tag and convert any "+"
-      # to "-".
       - name: Generate Docker image version tag
```
Comment (chainchad) [inline .github/workflows/publish.yml]: I took a quick look at our recent [tags](https://github.com/smartcontractkit/ccip/tags) (going back a month or so ago) and it looks like we're not using semver build metadata (denoted by a `+` symbol in the tag) so I think we can remove this step entirely 👍

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -121,26 +75,16 @@ jobs:
...
-          tags: ${{ steps.docker_meta.outputs.tags }}
-          labels: ${{ steps.docker_meta.outputs.labels }}
-          build-args: |
-            CHAINLINK_USER=chainlink
+          publish: true
+          aws-role-to-assume: ${{ secrets.AWS_OIDC_IAM_ROLE_PROD_PUBLISH_ARN }}
+          aws-role-duration-seconds: ${{ secrets.AWS_ROLE_DURATION_SECONDS }}
+          aws-region: ${{ env.AWS_REGION }}
+          ecr-hostname: ${{ env.AWS_ECR_REPO_PUBLIC_REGISTRY }}
+          ecr-image-name: ${{ env.AWS_ECR_REPO_PUBLIC_REGISTRY }}/w0i8p0z9/chainlink-ccip
```
Comment (chainchad) [inline .github/workflows/publish.yml]:
```suggestion
          ecr-image-name: w0i8p0z9/chainlink-ccip
```

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -121,26 +75,16 @@ jobs:
...
+          sign-images: false
+          dockerfile: ./core/chainlink.Dockerfile
+          git-commit-sha: ${{ steps.generate-docker-image-tag.outputs.tag }}
```
Comment (chainchad) [inline .github/workflows/publish.yml]:
```suggestion
```
I'd remove this entirely, the default is fine. If we did want to use it, we could pass in a git shasum but we wouldn't use the tag here.

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -121,26 +75,16 @@ jobs:
...
-      - name: Docker meta
-        id: docker_meta
-        uses: docker/metadata-action@507c2f2dc502c992ad446e3d7a5dfbe311567a96 # v4.3.0
+      - name: Build and publish chainlink image
+        uses: ./.github/actions/build-sign-publish-chainlink
         with:
-          images: ${{ env.AWS_ECR_REPO_PUBLIC_REGISTRY }}/w0i8p0z9/chainlink-ccip
-          tags: |
-            ${{ steps.generate-docker-image-tag.outputs.tag }}
-      - name: Fetch operator-ui
```
Comment (chainchad) [inline .github/workflows/publish.yml:131]: Making sure that it's OK to remove this?

**Review by chainchad — CHANGES_REQUESTED** [review]
Nice simplification but had some feedback/changes.

File: .github/workflows/publish.yml [hunk 5]
```diff
@@ -28,46 +28,19 @@ jobs:
...
-          tags: ${{ steps.docker_meta.outputs.tags }}
-          labels: ${{ steps.docker_meta.outputs.labels }}
-          build-args: |
-            CHAINLINK_USER=chainlink
+            publish: true
+            aws-role-to-assume: ${{ secrets.AWS_OIDC_IAM_ROLE_PROD_PUBLISH_ARN }}
+            aws-role-duration-seconds: ${{ secrets.AWS_ROLE_DURATION_SECONDS }}
+            aws-region: ${{ secrets.AWS_REGION }}
+            ecr-hostname: ${{ secrets.AWS_ECR_REPO_URL }}
+            ecr-image-name: ${{ secrets.AWS_ECR_REPO_URL }}
```
Comment (momentmaker) [inline .github/workflows/publish.yml]: this would be the image name and tag no?

File: .github/workflows/publish.yml [hunk 6]
```diff
@@ -121,26 +75,16 @@ jobs:
...
-        with:
-          context: .
-          push: true
-          file: ./core/chainlink.Dockerfile
-          tags: ${{ steps.docker_meta.outputs.tags }}
-          labels: ${{ steps.docker_meta.outputs.labels }}
-          build-args: |
-            CHAINLINK_USER=chainlink
+          publish: true
+          aws-role-to-assume: ${{ secrets.AWS_OIDC_IAM_ROLE_PROD_PUBLISH_ARN }}
```
Comment (momentmaker) [inline .github/workflows/publish.yml:72]: @chainchad I think we need a new role for `sdlc` for this right?

File: .github/workflows/publish.yml (hunk 5 above)
Comment (AnieeG) [reply .github/workflows/publish.yml]: With the previous docker_meta action this was used as image name-
```
  with:
          images: ${{ secrets.AWS_ECR_REPO_URL }}
```
Ported from that. Should I keep it blank ?

File: .github/workflows/publish.yml (hunk 6 above)
Comment (AnieeG) [reply .github/workflows/publish.yml:72]: This is the image published to prod separate from the image built for QA ecr. Do we still need sdlc role?

Comment (chainchad) [reply .github/workflows/publish.yml:72]: I think we can use the existing role in `secrets.AWS_OIDC_IAM_ROLE_PROD_PUBLISH_ARN` here.

File: .github/workflows/publish.yml (hunk 5 above)
Comment (chainchad) [reply .github/workflows/publish.yml]: Should be the same as this [suggestion](https://github.com/smartcontractkit/ccip/pull/392/files#r1432732054), I think.

File: .github/workflows/publish.yml (hunk 4 above)
Comment (AnieeG) [reply .github/workflows/publish.yml:131]: This is now built as part of `make generate `and committed to the repo. There is no need to build it again here

File: .github/workflows/publish.yml [hunk 7]
```diff
@@ -28,46 +28,18 @@ jobs:
...
-          file: ./core/chainlink.Dockerfile
-          tags: ${{ steps.docker_meta.outputs.tags }}
-          labels: ${{ steps.docker_meta.outputs.labels }}
-          build-args: |
-            CHAINLINK_USER=chainlink
+            publish: true
+            aws-role-to-assume: ${{ secrets.AWS_OIDC_IAM_ROLE_PROD_PUBLISH_ARN }}
+            aws-role-duration-seconds: ${{ secrets.AWS_ROLE_DURATION_SECONDS }}
+            aws-region: ${{ secrets.AWS_REGION }}
+            ecr-hostname: ${{ secrets.AWS_ECR_REPO_URL }}
```
Comment (chainchad) [inline .github/workflows/publish.yml:39]:
```suggestion
            ecr-hostname: ${{ secrets.AWS_ECR_REPO_URL }}
            ecr-image-name: w0i8p0z9/chainlink-ccip
```

**Review by chainchad — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 7 above)
Comment (chainchad) [inline .github/workflows/publish.yml:39]: We also need to potentially pass in the correct ecr-image-name.

**Review by chainchad — APPROVED** [review]

**Review by momentmaker — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 8]
```diff
@@ -6,7 +6,7 @@ on:
       - "v*"
     branches:
       - ccip-develop
-      - deployment-test
+      - "release/**"
```
Comment (mateusz-sekara) [inline .github/workflows/publish.yml]: Releng docs state that we should use `ccip-release/*` branch naming cc @chainchad

**Review by RensR — APPROVED** [review]

File: .github/workflows/publish.yml (hunk 8 above)
Comment (chainchad) [reply .github/workflows/publish.yml]: Good point. Should we update our docs to match release/** instead? IIRC there was talk about migrating /ccip code back into the /chainlink repo which would have necessitated that ccip- prefix. But if that's not in the cards, maybe we use release/** and update the docs?

Comment (mateusz-sekara) [reply .github/workflows/publish.yml]: I'm open to any solution as long as it's aligned with the docs and the current state. At this moment, we have all possible options in use 
image

Comment (chainchad) [reply .github/workflows/publish.yml]: Thanks, tagged you all in a separate PR to update the docs.

**Pull request merged**